lib/puppet/application/lookup.rb

```diff
@@ -339,6 +339,7 @@ def generate_scope
 
     Puppet[:code] = 'undef' unless options[:compile]
     compiler = Puppet::Parser::Compiler.new(node)
+    compiler.loaders # ensures that Pcore et.al. is initialized
```

Comment:
Doesn't this mean that you would always need to follow a call to `Compiler.new` with this call? 
And then, doesn't that mean that this potentially break other use cases of compiler? Like catalog preview (to mention one of ours) or puppet debug from the community?